src/main/java/seedu/trackermon/logic/parser/SortCommandParser.java

```diff
@@ -31,48 +47,143 @@ public SortCommand parse(String args) throws ParseException {
+        Comparator<Show> ratingComparator = new RatingComparator();
+        Comparator<Show> tagComparator = new TagComparator();
 
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_SORT_NAME_ACS,
-                        PREFIX_SORT_STATUS_ACS, PREFIX_SORT_NAME_DES,
-                        PREFIX_SORT_STATUS_DES, PREFIX_SORT_STATUS_ORD);
-
-        if (arePrefixesPresent(argMultimap, PREFIX_SORT_NAME_ACS)) {
-            isNamePresent = true;
-        } else if (arePrefixesPresent(argMultimap, PREFIX_SORT_NAME_DES)) {
-            isNamePresent = true;
-            nameComparator = nameComparator.reversed();
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME,
+                        PREFIX_STATUS, PREFIX_RATING, PREFIX_TAG, PREFIX_SORT_ORDER);
+
+        order = new HashMap<>();
+        //Initialise order map
+        order.put(nameComparator, NO_VALUE);
+        order.put(statusComparator, NO_VALUE);
+        order.put(ratingComparator, NO_VALUE);
+        order.put(tagComparator, NO_VALUE);
+
+        nameComparator = putIntoMap(nameComparator, PREFIX_NAME, argMultimap);
+        statusComparator = putIntoMap(statusComparator, PREFIX_STATUS, argMultimap);
+        ratingComparator = putIntoMap(ratingComparator, PREFIX_RATING, argMultimap);
+        tagComparator = putIntoMap(tagComparator, PREFIX_TAG, argMultimap);
+
+        HashMap<Comparator<Show>, String> comparatorString = new HashMap<>();
+        comparatorString.put(nameComparator, VALUE_ORDER_NAME);
+        comparatorString.put(statusComparator, VALUE_ORDER_STATUS);
+        comparatorString.put(ratingComparator, VALUE_ORDER_RATING);
+        comparatorString.put(tagComparator, VALUE_ORDER_TAG);
+        reorderMap(argMultimap, comparatorString);
+
+        //sort the hashmap
+        Map<Comparator<Show>, Integer> sortedOrder = order.entrySet().stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        Map.Entry::getValue, (oldValue, newValue) -> oldValue, LinkedHashMap::new));
+
+        Comparator<Show> comparator = buildComparator(sortedOrder);
+
+        //no prefixes are present, sort by name ascending
+        if (comparator == null) {
+            return new SortCommand(nameComparator);
         }
 
-        if (arePrefixesPresent(argMultimap, PREFIX_SORT_STATUS_ACS)) {
-            isStatusPresent = true;
-        } else if (arePrefixesPresent(argMultimap, PREFIX_SORT_STATUS_DES)) {
-            isStatusPresent = true;
-            statusComparator = statusComparator.reversed();
+        return new SortCommand(comparator);
+    }
+
+    /**
+     * Returns a comparator then chain the previous comparator to a new one if the previous is not null.
+     * @param comparator previous comparator to build upon.
+     * @param entry contain the new comparator.
+     * @return a comparator build upon the previous comparator.
```

Comment:
Minor grammatical error here
`a comparator build` should be replaced with `a comparator built`

Reply:
ok